### PR TITLE
feat: add NATS reconnect handling

### PR DESCRIPTION
**Pull request by Aenimus**
## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: router/core/executor.go [hunk 1]
```diff
@@ -189,19 +189,37 @@ func (b *ExecutorConfigurationBuilder) Build(ctx context.Context, opts *Executor
 }
 
 func buildNatsOptions(eventSource config.NatsEventSource, logger *zap.Logger) ([]nats.Option, error) {
-
 	opts := []nats.Option{
```
Comment (jensneuse) [inline router/core/executor.go:192]: I'd add `nats. AllowReconnect(true)`

**Review by jensneuse — APPROVED** [review]
one minor comment

File: router/core/executor.go (hunk 1 above)
Comment (StarpTech) [reply router/core/executor.go:192]: To ensure the default? Is is enabled by default

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**